pyk/src/pyk/kast/inner.py

```diff
@@ -885,12 +885,14 @@ def collect(callback: Callable[[KInner], None], kinner: KInner) -> None:
         callback: Function with the side effect of collecting desired information at each AST node.
         kinner: The term to traverse.
     """
+    subterms = [kinner]
 
-    def f(kinner: KInner) -> KInner:
-        callback(kinner)
-        return kinner
+    for term in subterms:
+        subterms.extend(term.terms)  # noqa: B909
```

Comment:
`B909` here is not a false positive:

```
Found a mutation of a mutable loop iterable inside the loop body.
```

But I think this is acceptable here to avoid having to maintain two lists.

Reply:
(It would be easier to do a top-down traversal, but since the docstring mentions traversal order I didn't want to change it.)